Tests/ResultTests/ResultTests.swift

```diff
@@ -120,21 +120,26 @@ final class ResultTests: XCTestCase {
 	}
 
 	func testMaterializeProducesSuccesses() {
-		let result1: Result<String, AnyError> = materialize(try tryIsSuccess("success"))
+		let result1: Result<String, AnyError> = Result(try tryIsSuccess("success"))
 		XCTAssert(result1 == success)
 
-		let result2: Result<String, AnyError> = materialize { try tryIsSuccess("success") }
+		let result2: Result<String, AnyError> = Result { try tryIsSuccess("success") }
 		XCTAssert(result2 == success)
 	}
 
 	func testMaterializeProducesFailures() {
-		let result1: Result<String, AnyError> = materialize(try tryIsSuccess(nil))
+		let result1: Result<String, AnyError> = Result(try tryIsSuccess(nil))
 		XCTAssert(result1.error == error)
 
-		let result2: Result<String, AnyError> = materialize { try tryIsSuccess(nil) }
+		let result2: Result<String, AnyError> = Result { try tryIsSuccess(nil) }
 		XCTAssert(result2.error == error)
 	}
 
+	func testMaterializeInferrence() {
+		let result = Result { try tryIsSuccess(nil) }
+		XCTAssert((type(of: result) as Any.Type) is Result<() throws -> String, AnyError>.Type)
```

Comment:
Do note that the inferred `T` type is the same as `materialize`, just that it would be ambiguous due to having both the `AnyError` and `NSError` overloads.